src/containers/ProfilePage/ProfilePage.js

```diff
@@ -328,6 +335,15 @@ export const ProfilePageComponent = props => {
         params={{ missingAccessRight: NO_ACCESS_PAGE_USER_PENDING_APPROVAL }}
       />
     );
+  } else if (!isPreview && hasNoViewingRightsOnPrivateMarketplace && !isCurrentUser) {
```

Comment:
I started to think that this "if - else if" chain does not work correctly.

```js
  if (!isDataLoaded) {
    // return etc.
  } else if (!isPreview && isNotFoundError(userShowError)) {
    // return etc.
  } else if (!isPreview && (isUnauthorizedOnPrivateMarketplace || hasUserPendingApprovalError)) {
    // return etc.
  } else if (!isPreview && !isCurrentUser && (hasNoViewingRightsOnPrivateMarketplace || isErrorNoViewingRights(userShowError))) {
    // TODO 
    // - Is it possible to create isErrorNoViewingRights error check?
    // - SSR: should only end up here if isErrorNoViewingRights is thrown (hasNoViewingRightsOnPrivateMarketplace contains assumption of currentUser entity)
    // return etc.
  } else if (!isPreview && isForbiddenError(userShowError)) {
    // The generic 403 Forbidden needs to come after more specific 403s aka hasUserPendingApprovalError (and potential isErrorNoViewingRights that doesn't exist yet)
    // return etc.
  } else if (isPreview && mounted && !isCurrentUser) {
    // return etc.
  } else if ((isPreview || isPrivateMarketplace) && !mounted) {
    // return etc.
  }
  
  // This is rendering normal profile page (not preview for pending-approval)
  return (
```

Reply:
Does SSR apply on a private marketplace?

I tested this approach anyway now to see if it's possible to create isErrorNoViewingRights here. My initial implementation was to not dispatch showUser at all when `canFetchOwnProfileOnly`, but I tested adding the dispatch in the promises array to catch the error and use it in this condition. 

If I add `showUser` dispatch and remove the `hasNoViewingRightsOnPrivateMarketplace` check from `isDataLoaded`, (so loads the page once `user` or `userShowError` exists) the `userShowError` has time to propagate before the page loads, so that I could add an `isErrorNoViewingRights` function. (This means that the isForbiddenError check in the current implementation is unnecessary, because without the showUser dispatch the userShowError always remains null).

However, if SSR doesn't apply on private marketplaces, this would only apply client-side. It would not change the behavior, and instead it would add one extra SDK call that we only make to get the error, because the `canFetchOwnProfileOnly` condition already has different dispatches going on (own listings vs all listings).

So either way works!

Reply:
Condition updated to 

`  } else if (
    (!isPreview && hasNoViewingRightsOnPrivateMarketplace && !isCurrentUser) ||
    isErrorNoViewingPermission(userShowError)
  ) {`

and isErrorNoViewingPermission added.

I also added similar handling to search pages:

`  } else if ((isPrivateMarketplace && hasNoViewingRightsUser) || hasNoViewingRightsError) {`